envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/server/CachedProtoResourcesSerializer.kt

```diff
@@ -7,13 +7,20 @@ import com.google.protobuf.Message
 import io.envoyproxy.controlplane.server.serializer.ProtoResourcesSerializer
 import io.micrometer.core.instrument.MeterRegistry
 import io.micrometer.core.instrument.binder.cache.GuavaCacheMetrics
+import pl.allegro.tech.servicemesh.envoycontrol.utils.noopTimer
 import java.util.function.Supplier
 
 internal class CachedProtoResourcesSerializer(
-    private val meterRegistry: MeterRegistry,
-    private val reportMetrics: Boolean
+    meterRegistry: MeterRegistry,
+    reportMetrics: Boolean
 ) : ProtoResourcesSerializer {
 
+    private val serializeTimer = if (reportMetrics) {
```

Comment:
I like this, makes code cleaner 👍

Reply:
Since we'll be using it mostly with feature flags I think we can even extract this block as a function instead of just `noopTimer`. Something like this:
```kotlin
fun timer(metricName: String, reportMetric: Boolean) =
// or
fun timer(metricName: String, reportMetric: () -> Boolean)
    if (report) {
        meterRegistry.timer(metricName)
    } else {
        noopTimer
    }
```
Just an idea.

Reply:
I didn't do that only because I wanted to skip unnecessary string interpolation and allocation for `metricName` (which is dynamic in some cases). I know - paranoid level of optimisations.

Reply:
I think it would be jitted.

Reply:
In this case yes, now I think i could indeed do as you proposed. I confused this with other place  where the metric name is dynamic (it includes serviceName)